File: TheBigEvent/src/TheBigEvent/Services/EventService.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TheBigEvent.DAL;

namespace TheBigEvent.Services
{
    public class EventService
    {
        readonly EventLink _eLink;
        public EventService(string connectionString)
        {
            _eLink = new EventLink(connectionString);
        }
        public Result<IEnumerable<Event>> getEvent()
        {
            return Result.Success(Status.Ok, _eLink.GetAllEvent());
        }
        public Result<IEnumerable<Event>> getEventById(string id)
        {
            return Result.Success(Status.Ok, _eLink.GetAllEventById(id));
        }
        public Result<IEnumerable<Event>> getEventSalle(int id)
        {
            return Result.Success(Status.Ok, _eLink.GetEventSalle(id));
        }
        public Result<IEnumerable<Event>> getEventDeco(int id)
        {
            return Result.Success(Status.Ok, _eLink.GetEventDeco(id));
        }
        public Result<IEnumerable<Event>> getEventTraiteur(int id)
        {
            return Result.Success(Status.Ok, _eLink.GetEventTraiteur(id));
        }
        public Result<IEnumerable<Event>> getEventProById(string id)
        {
            return Result.Success(Status.Ok, _eLink.GetAllEventProById(id));
        }
        public void addEvent(string _NomEvent, string _Localisation, int _MenuId, int _SalleId, int _TraiteurId, int _DecoId, int _UserId, int NbInvite, int Prix, DateTime Dates)
        {
            _eLink.AddEvent( _NomEvent, _Localisation, _MenuId, _SalleId, _TraiteurId,_DecoId,_UserId, NbInvite, Prix, Dates);
        }
        public Result<IEnumerable<Event>> getEventByIdPT(int id)
        {
            return Result.Success(Status.Ok, _eLink.GetEventByIdT(id));
        }
        public Result<IEnumerable<Event>> getEventByIdPD(int id)
        {
            return Result.Success(Status.Ok, _eLink.GetEventByIdD(id));
        }
        public Result<IEnumerable<Event>> getEventByIdPS(int id)
        {
            return Result.Success(Status.Ok, _eLink.GetEventByIdS(id));
        }
        public Result<IEnumerable<Event>> getEventByIdPM(int id)
        {
            return Result.Success(Status.Ok, _eLink.GetEventByIdM(id));
        }

        public void validT(int id, int value)
        {
            _eLink.validT(id, value);
            return;
        }
        public void validD(int id, int value)
        {
            _eLink.validD(id, value);
            return;
        }
        public void validS(int id, int value)
        {
            _eLink.validS(id, value);
            return;
        }

        public void UpdateTraiteurIdbynull(int id)
        {
            _eLink.UpdateTraiteurIdbynull(id);
            return;
        }

        public void UpdateMenuIdbynull(int id)
        {
            _eLink.UpdateMenuIdbynull(id);
            return;
        }

        public void UpdateSalleIdbynull(int id)
        {
            _eLink.UpdateSalleIdbynull(id);
            return;
        }

        public void UpdateDecoIdbynull(int id)
        {
            _eLink.UpdateDecoIdbynull(id);
            return;
        }



        public void DeleteEvent(int _EventId)
        {
            _eLink.Delete(_EventId);
        }
    }
}

File: TheBigEvent/src/TheBigEvent/App/src/services/EventService.js
import { getAsync, postAsync, putAsync, deleteAsync, getAsyncSelect } from '../helpers/req'
import AuthService from './auth'

const endpoint = "/api/Event";

class EventService {
    constructor() {
    }

// On crée un évènement

    async createEventAsync(model) 
    {
        return await postAsync(endpoint, 'createEvent/', AuthService.accessToken, model);
    }

// On supprime un évènement

    async deleteEvent(Id)
    {
        return await deleteAsync(endpoint, 'deleteEvent/' + Id, AuthService.accessToken);
    }

    async selectGetAsync(method){
        return await getAsyncSelect(endpoint, method, AuthService.accessToken)
    }

    async getEventbyidPD(id)
    {
        return await getAsyncSelect(endpoint, 'event/deco/' + id, AuthService.accessToken)        
    }
    async getEventbyidPT(id)
    {
        return await getAsyncSelect(endpoint, 'event/traiteur/' + id, AuthService.accessToken)        
    }
    async getEventbyidPS(id)
    {
        return await getAsyncSelect(endpoint,'event/salle/' + id, AuthService.accessToken)        
    }
    async getEventbyidPM(id)
    {
        return await getAsyncSelect(endpoint,'event/menu/' + id, AuthService.accessToken)        
    }

    async validT(model) 
    {
        return await postAsync(endpoint, 'validT', AuthService.accessToken, model);
    }

    async validD(model) 
    {
        return await postAsync(endpoint, 'validD', AuthService.accessToken, model);
    }

    async validS(model) 
    {
        return await postAsync(endpoint, 'validS', AuthService.accessToken, model);
    }


    async UpdateSalleIdbynull(eventId)
    {
        return await postAsync(endpoint, 'UpdateSalleIdbynull/' + eventId, AuthService.accessToken);
    }

    async UpdateTraiteurIdbynull(eventId)
    {
        return await postAsync(endpoint, 'UpdateTraiteurIdbynull/' + eventId, AuthService.accessToken);
    }

    async UpdateDecoIdbynull(eventId)
    {
        return await postAsync(endpoint, 'UpdateDecoIdbynull/' + eventId, AuthService.accessToken);
    }

    async UpdateMenuIdbynull(eventId)
    {
        return await postAsync(endpoint, 'UpdateMenuIdbynull/' + eventId, AuthService.accessToken);
    }


}

export default new EventService()